keras/utils/data_utils.py

```diff
@@ -365,12 +365,19 @@ def on_epoch_end(self):
         pass
 
 # Global variables to be shared across processes
-_SHARED_SEQUENCE = None
-_MANAGER = multiprocessing.Manager()
-_SHARED_DICT = _MANAGER.dict()
+_SHARED_SEQUENCES = {}
+_MANAGERS = {}
+_SHARED_DICTS = {}
 
 
-def get_index(i):
+def _initialize_globals(uuid):
+    """Initialize the inner dictionary to manage processes."""
+    global _SHARED_DICTS, _MANAGERS
+    _MANAGERS[uuid] = multiprocessing.Manager()
+    _SHARED_DICTS[uuid] = _MANAGERS[uuid].dict()
+
+
+def get_index(uuid,i):
     """Get the value from the Sequence at index `i`.
```

Comment:
This description it too terse with respect to the unexpected signature (`uuid, i`). Please add a comment about the shared sequences system and how they are addressed by an index.